.github/workflow_scripts/run_benchmark.sh

```diff
@@ -14,31 +14,44 @@ source $(dirname "$0")/env_setup.sh
-    mv "./results/$master_cleaned_file" "./results/$new_master_cleaned_file"
+    if [ $MODULE != "multimodal" ]; then
+        master_cleaned_file=$(aws s3 ls s3://autogluon-ci-benchmark/cleaned/$MODULE/master/latest/ | awk '{print $NF}')
+        new_master_cleaned_file="master_${master_cleaned_file}"
+        aws s3 cp --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/master/latest/ ./results
+        mv "./results/$master_cleaned_file" "./results/$new_master_cleaned_file"
+    else
+        master_cleaned_file=$(aws s3 ls s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/master/latest/ | awk '{print $NF}')
+        new_master_cleaned_file="master_${master_cleaned_file}"
+        aws s3 cp --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/master/latest/ ./results
+        mv "./results/$master_cleaned_file" "./results/$new_master_cleaned_file"
+    fi
 fi
 
 python CI/bench/evaluate.py --config_path ./ag_bench_runs/$MODULE/ --module_name $MODULE --time_limit $TIME_LIMIT --branch_name $BRANCH_OR_PR_NUMBER
 
 for file in ./results/*; do
     # Check if the file does not start with "master"
-    if [[ "$(basename "$file")" != "master"* ]]
-    then
+    if [[ "$(basename "$file")" != "master"* ]] && [ $MODULE != "multimodal" ]; then
         aws s3 cp "$file" "s3://autogluon-ci-benchmark/cleaned/$MODULE/$BRANCH_OR_PR_NUMBER/$SHA/$(basename "$file")"
         aws s3 rm --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/$BRANCH_OR_PR_NUMBER/latest/
         aws s3 cp "$file" s3://autogluon-ci-benchmark/cleaned/$MODULE/$BRANCH_OR_PR_NUMBER/latest/$(basename "$file")
-    else
+    elif [[ "$(basename "$file")" == "master"* ]] && [ $MODULE != "multimodal" ]; then
         aws s3 cp "$file" "s3://autogluon-ci-benchmark/cleaned/$MODULE/master/$SHA/$(basename "$file")"
         aws s3 rm --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/master/latest/
         aws s3 cp "$file" s3://autogluon-ci-benchmark/cleaned/$MODULE/master/latest/$(basename "$file")
+    elif [[ "$(basename "$file")" != "master"* ]] && [ $MODULE == "multimodal" ]; then
+        aws s3 cp "$file" "s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/$BRANCH_OR_PR_NUMBER/$SHA/$(basename "$file")"
+        aws s3 rm --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/$BRANCH_OR_PR_NUMBER/latest/
+        aws s3 cp "$file" s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/$BRANCH_OR_PR_NUMBER/latest/$(basename "$file")
+    elif [[ "$(basename "$file")" == "master"* ]] && [ $MODULE == "multimodal" ]; then
+        aws s3 cp "$file" "s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/master/$SHA/$(basename "$file")"
+        aws s3 rm --recursive s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/master/latest/
+        aws s3 cp "$file" s3://autogluon-ci-benchmark/cleaned/$MODULE/$BENCHMARK/master/latest/$(basename "$file")
```

Comment:
Because for tabular and timeseries we store it in bucket-name->cleaned->{tabular/timeseries}->master.
We don't have different types of data like text, text-tabular, image as we do for multimodal.
Hence, for better readability and segregation I have it as the following for multimodal:
bucket-name->cleaned->multimodal->{text/text-tabular/image}->master, so we can know which benchmark was run.

Reply:
That makes sense. Also I think you can simplify the logic by decoupling the two conditions as below:
```
if [[ "$(basename "$file")" == "master"* ]]; then
          BRANCH_NAME="master"
else
          BRANCH_NAME=$BRANCH_OR_PR_NUMBER
fi
if [[ $MODULE == "multimodal" ]]; then
          SUB_FOLDER="$BENCHMARK/"
else
          SUB_FOLDER=""
fi
```
Then you don't have to repeat similar commands for 4 times.

Reply:
We also have to add the check to see if the file does not start with master everytime because we do not want to copy that result as it is the master branch's cleaned result and not the PR branch and with the above code it will not do that, hence we have these many checks

Reply:
Try something like below? The code should be simplified to reduce redundancy and improve readability:

```
CLEANED_PATH="s3://autogluon-ci-benchmark/cleaned/$MODULE"
EVALUATION_PATH="s3://autogluon-ci-benchmark/evaluation/$MODULE"
BRANCH_NAME="master"
if [[ "$(basename "$file")" != "master"* ]]; then
    BRANCH_NAME="$BRANCH_OR_PR_NUMBER"
fi

if [ $MODULE == "multimodal" ]; then
    CLEANED_PATH="$CLEANED_PATH/$BENCHMARK"
    EVALUATION_PATH="$EVALUATION_PATH/$BENCHMARK"
fi

aws s3 cp "$file" "$CLEANED_PATH/$BRANCH_NAME/$SHA/$(basename "$file")"
aws s3 rm --recursive "$CLEANED_PATH/$BRANCH_NAME/latest/"
aws s3 cp "$file" "$CLEANED_PATH/$BRANCH_NAME/latest/$(basename "$file")"

if [[ "$(basename "$file")" != "master"* ]]; then
    aws s3 cp --recursive ./evaluate "$EVALUATION_PATH/$BRANCH_NAME/$SHA/"
    aws s3 rm --recursive "$EVALUATION_PATH/$BRANCH_NAME/latest/"
    aws s3 cp --recursive ./evaluate "$EVALUATION_PATH/$BRANCH_NAME/latest/"
fi
```